File: rb/features.rb
# CatchdomsSecurity SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module CatchdomsSecurityFeatures
  def self.make_feature(name)
    case name
    when "base"
      CatchdomsSecurityBaseFeature.new
    when "test"
      CatchdomsSecurityTestFeature.new
    else
      CatchdomsSecurityBaseFeature.new
    end
  end
end

File: py/features.py
# CatchdomsSecurity SDK feature factory

from feature.base_feature import CatchdomsSecurityBaseFeature
from feature.test_feature import CatchdomsSecurityTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CatchdomsSecurityBaseFeature(),
        "test": lambda: CatchdomsSecurityTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
